apps/saleor-app-checkout/frontend/components/templates/ChannelDetails/ChannelDetails.tsx

```diff
@@ -95,8 +95,12 @@ const ChannelDetails: React.FC<ChannelDetailsProps> = ({
   };
 
   const handleSubmit = (flattenedSettings: Record<string, string>) => {
+    const settingsWithValues = Object.entries(flattenedSettings).reduce(
+      (prev, [key, value]) => ({ ...prev, ...(value !== "" && { [key]: value }) }),
```

Comment:
Could we add this comment? It took me a moment to figure out what this `.reduce` does
```suggestion
      // filters entries without values
      (prev, [key, value]) => ({ ...prev, ...(value !== "" && { [key]: value }) }),
```